### PR TITLE
[Feature] Build Physician Assessment Edit Modal

### PR DESCRIPTION
**Pull request by j-bunk**
## Notion ticket link

[Build Physician Assessment edit modal](https://www.notion.so/uwblueprintexecs/Build-Physician-Assessment-edit-modal-337e42efe73f4bd59ec1a98b2415ea38)

## Implementation description
Added Physician Assessment Edit Modal and mobility aids section to card + modal

- [x] Build edit modal
- [x] Add `UPDATE_PHYSICIAN_ASSESSMENT` mutation
- [x] Hook up modal to card
  - [x] Hook up modal + add mutation call
  - [x] Add mobility section to card (missed in previous PR)
  - [x] Add `permitType` to `NewApplication` table (confirm with @OustanDing)
- [x] Add mobility aids section to form component (unrelated but must have been missed when the form component was built)
  - [x] Build section
  - [x] Update db to include `CRUTCHES` as a mobility aid, and also add `otherMobilityAids` field
- [x] Address NOTETOSELF comment(s)

## Demo

https://user-images.githubusercontent.com/51224641/167760884-11e99933-c7b3-4bdc-9db3-aca293aaa7b7.mov

## Notes
* I noticed that the mobility aids section was missing from the Physician Assessment card and form components and added it to them in this PR
* Form validation needs to be added and the edit modal will need to be refactored accordingly (needs to be wrapped in a Formik component). I think this can be addressed in the form validation PR once this is merged in.
* Will merge after https://github.com/uwblueprint/richmond-centre-for-disability/pull/202 is merged

## Checklist
- [x] My PR name is descriptive, is in imperative tense and starts with one of the following: `[Feature]`,`[Improvement]` or `[Fix]`,
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the RCD team on GitHub, or specific people who are associated with this ticket

### CODE REVIEW

File: components/admin/requests/additional-questions/EditModal.tsx
Comment (j-bunk) [inline components/admin/requests/additional-questions/EditModal.tsx:65]: Unrelated to this PR

File: components/admin/requests/physician-assessment/Card.tsx
Comment (j-bunk) [inline components/admin/requests/physician-assessment/Card.tsx:187]: I must have missed this when building the card. It's in some of the specs and missing in others, I'll confirm with designers if this section is conditionally rendered

File: components/admin/requests/physician-assessment/Form.tsx
Comment (j-bunk) [inline components/admin/requests/physician-assessment/Form.tsx:165]: Lmk if there's a cleaner way to handle this. The current approach is to use a state `mobilityAidsRequired` to determine whether or not to render the Other mobility aids text area. If the user selects that they do not have mobility aids the if statement is hit and the value of mobility aid is set to `[]`.

File: lib/applications/schema.ts
Comment (j-bunk) [inline lib/applications/schema.ts:755]: I think `permitType` needs to be included to allow users to change the `permitType` using the `EditModal` but it seems to have been purposely omitted here (there's a comment on line 748 saying to omit it).

image

File: prisma/schema.prisma
Comment (j-bunk) [inline prisma/schema.prisma:249]: Added `otherMobilityAids` to match the design

image

File: prisma/schema.prisma
Comment (j-bunk) [inline prisma/schema.prisma:252]: See above https://github.com/uwblueprint/richmond-centre-for-disability/pull/192/files#r869824458

File: prisma/schema.prisma
Comment (j-bunk) [inline prisma/schema.prisma:446]: Added `CRUTCHES` and `OTHERS` to match the design

image

File: tools/admin/requests/additional-questions.ts
Comment (j-bunk) [inline tools/admin/requests/additional-questions.ts:59]: Unrelated to this PR

**Review by carelynntsai — APPROVED** [review]
😍

File: components/admin/requests/physician-assessment/Card.tsx
Comment (j-bunk) [reply components/admin/requests/physician-assessment/Card.tsx]: Resolved!

File: components/admin/requests/physician-assessment/Card.tsx
Comment (j-bunk) [inline components/admin/requests/physician-assessment/Card.tsx:207]: Confirming with designers if they want this (see slack thread https://uwblueprint.slack.com/archives/C021S4WGHBR/p1652067821964179)

**Review by OustanDing — CHANGES_REQUESTED** [review]

File: components/admin/requests/physician-assessment/Form.tsx
Comment (j-bunk) [reply components/admin/requests/physician-assessment/Form.tsx]: Addressed!

File: components/admin/requests/physician-assessment/EditModal.tsx
Comment (j-bunk) [reply components/admin/requests/physician-assessment/EditModal.tsx]: Addressed!

File: prisma/schema.sql
Comment (j-bunk) [reply prisma/schema.sql]: Addressed in latest commit!

File: lib/applications/schema.ts
Comment (j-bunk) [reply lib/applications/schema.ts:755]: Would you mind clarifying here? Is there anything I should look out for 👀

Comment (j-bunk) [reply lib/applications/schema.ts:777]: @OustanDing bump^

**Comment by j-bunk** [comment]
@OustanDing Now that https://github.com/uwblueprint/richmond-centre-for-disability/pull/202 is merged in, this PR can be merged into the larger form validation PR https://github.com/uwblueprint/richmond-centre-for-disability/pull/158.

I've addressed all your PR feedback except for https://github.com/uwblueprint/richmond-centre-for-disability/pull/192#discussion_r870883630, once that's resolved I'll merge this PR in

File: lib/applications/schema.ts
Comment (OustanDing) [reply lib/applications/schema.ts:777]: Yes, to clarify, your change here is correct, so no further action is needed in this file. What I meant by backwards compatibility is that we should ensure that any other API calls that depend on this input schema should still work.

Comment (j-bunk) [reply lib/applications/schema.ts:777]: Sweet, tested the api the relevant api calls and everything looks good!

**Comment by j-bunk** [comment]
@OustanDing address the comment! PR should be good to merge if things look good to you!

**Review by OustanDing — APPROVED** [review]

**Pull request merged**
